distribute/distribute.sh

```diff
@@ -1,193 +1,288 @@
-if [ "$answer" != "n" ]
-then
-  pushd . > /dev/null
-  cp linux/*-pak ../
-  cd ..
-  # hide doc directory from checkinstall
-  # mv doc doctmp
-  test -f /usr/bin/visicut && { echo "error: please first uninstall visicut"; exit 1; }
-  fakeroot checkinstall --fstrans --reset-uid --type debian --install=no -y --pkgname visicut --pkgversion $VERSION --arch all --pkglicense LGPL --pkggroup other --pkgsource "http://visicut.org" --pkgaltsource "https://github.com/t-oster/VisiCut" --pakdir distribute/ --maintainer "'Thomas Oster <thomas.oster@rwth-aachen.de>'" --requires 'bash,openjdk-11-jre\|openjdk-17-jre,potrace' make install -e PREFIX=/usr > /dev/null || { echo "error"; exit 1; }
-  rm -rf *-pak
-  # mv doctmp doc
-  popd > /dev/null
+export VERSION
+
+if [ "${BUILD:-}" != "" ]; then
+    log "Building VisiCut JAR"
+    # TODO: make out-of-source builds possible
+	pushd "$project_root_dir"
+	make clean jar
+	popd
 fi
 
-echo "Dir: $(pwd)"
-echo "****************************************************************"
-echo "Arch Linux Version: For Building you must have pacman installed."
-echo "Build Arch Linux Version (Y/n)?"
-read answer || true
-if [ "$answer" != "n" ]
-then
-  cd linux
-  ARCHVERSION=$(echo $VERSION|sed "s#-#_#g")
-  cat PKGBUILD | sed "s#pkgver=VERSION#pkgver=$ARCHVERSION#g" > PKGBUILD-tmp
-  makepkg -p PKGBUILD-tmp > /dev/null || exit 1
-  mv *.pkg.tar.xz ../
-  echo "Cleaning up..."
-  rm -rf src pkg PKGBUILD-tmp
-  cd ..
+# we "ab"use the deployment directory to create the "visicut directory"
+# this directory contains all of VisiCut's binaries and resources
+# consider it a semi-portable bundle (a JVM is needed to make it complete)
+# the resulting directory can then be put into various types of packages
+visicut_dir="$(readlink -f "$distribute_dir"/visicut)"
+
+if [[ -d "$visicut_dir" ]]; then
+    log "Cleaning up old visicut directory in $visicut_dir"
+    rm -r "$visicut_dir"
 fi
 
-echo "Cleaning up..."
-rm -rf visicut
-echo "done."
+log "Creating visicut directory in $visicut_dir"
+
+mkdir "$visicut_dir"
+
+cp -v "$project_root_dir"/target/visicut-*-full.jar "$visicut_dir"/Visicut.jar
+cp -R -v "$distribute_dir"/files/* "$visicut_dir"/
+chmod +x "$visicut_dir"/{*.jar,VisiCut.*}
+
+cp -v "$project_root_dir"/README.md "$visicut_dir"/README
+cp -v "$project_root_dir"/{COPYING.LESSER,LICENSE} "$visicut_dir"/
+
+mkdir -p "$visicut_dir"/{inkscape_extension,illustrator_script}
+cp -v "$project_root_dir"/tools/inkscape_extension/{*.py,*.inx} "$visicut_dir"/inkscape_extension/
+cp -v "$project_root_dir"/tools/illustrator_script/*.scpt "$visicut_dir"/illustrator_script/
+
+# now that the visicut directory has been set up, we can perform platform-specific bundling
+for target in "$@"; do
+    log "Building for target $target"
+
+    build_dir="$(mktemp -d -t visicut-build-$target-XXXXXX)"
+    cleanup() {
+        [[ -d "$build_dir" ]] && rm -rf "${build_dir:?}"/
+    }
+    trap cleanup EXIT
+
+    download_and_extract_jdk() {
```

Comment:
I would suggest moving this function outside the for loop since it doesn't depend on the loop variables.